tests/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtensionTest.php

```diff
@@ -443,19 +455,7 @@ public function testFetchEagerWithNoForceEager()
 INNER JOIN o.compositeLabel label
 LEFT JOIN o.foo foo WITH o.bar = item.foo
 LEFT JOIN ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\DummyCar car WITH car.id = o.car
-WHERE o.item IN(
-    SELECT IDENTITY(o_2.item) FROM ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\CompositeRelation o_2
-    INNER JOIN o_2.compositeItem item_2
-    INNER JOIN o_2.compositeLabel label_2
-    LEFT JOIN o_2.foo foo_2 WITH o_2.bar = item_2.foo
-    WHERE item_2.field1 = :foo
-) AND o.label IN(
-    SELECT IDENTITY(o_2.label) FROM ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\CompositeRelation o_2
-    INNER JOIN o_2.compositeItem item_2
-    INNER JOIN o_2.compositeLabel label_2
-    LEFT JOIN o_2.foo foo_2 WITH o_2.bar = item_2.foo
-    WHERE item_2.field1 = :foo
-)
+WHERE item.field1 = :foo
```

Comment:
OK, I will take a look.

Reply:
Could you please tell me what do you want exactly to test, please ? Since the composite relation cannot have `ONE_TO_MANY`, the test case doesn't exist. If you mean a simple `ONE_TO_MANY`, it is tested in `testApplyCollectionCorrectlyReplacesJoinCondition` :)

Reply:
@soyuka Did you mean the following test case ? If yes, I can add it.
```php
 public function testMultipleOneToManyFetchEager()
    {
        $resourceMetadataFactoryProphecy = $this->prophesize(ResourceMetadataFactoryInterface::class);
        $resourceMetadataFactoryProphecy->create(DummyProduct::class)->willReturn(new ResourceMetadata(DummyProduct::class));

        $productClassMetadata = new ClassMetadataInfo(DummyProduct::class);
        $productClassMetadata->identifier = ['id'];
        $productClassMetadata->associationMappings = [
            'offers' => ['fetch' => ClassMetadataInfo::FETCH_EAGER, 'targetEntity' => DummyAggregateOffer::class, 'mappedBy' => 'product', 'type' => ClassMetadataInfo::ONE_TO_MANY],
            'relatedProducts' => ['fetch' => ClassMetadataInfo::FETCH_EAGER, 'targetEntity' => DummyProduct::class, 'mappedBy' => 'parent', 'type' => ClassMetadataInfo::ONE_TO_MANY],
        ];

        $em = $this->prophesize(EntityManager::class);
        $em->getExpressionBuilder()->shouldBeCalled()->willReturn(new Expr());
        $em->getClassMetadata(DummyProduct::class)->shouldBeCalled()->willReturn($productClassMetadata);

        $qb = new QueryBuilder($em->reveal());

        $qb->select('o')
            ->from(DummyProduct::class, 'p')
            ->innerJoin('p.offers', 'offers')
            ->innerJoin('p.relatedProducts', 'relatedProducts')
            ->where('offers.value = :offerValue')
            ->andWhere('relatedProducts.name = :productName')
            ->setParameter('offerValue', 1)
            ->setParameter('productName', 'foo');

        $queryNameGenerator = $this->prophesize(QueryNameGeneratorInterface::class);
        $queryNameGenerator->generateJoinAlias('p')->shouldBeCalled()->willReturn('p');
        $queryNameGenerator->generateJoinAlias('offers')->shouldBeCalled()->willReturn('offers');
        $queryNameGenerator->generateJoinAlias('relatedProducts')->shouldBeCalled()->willReturn('relatedProducts');

        $filterEagerLoadingExtension = new FilterEagerLoadingExtension($resourceMetadataFactoryProphecy->reveal(), false);
        $filterEagerLoadingExtension->applyToCollection($qb, $queryNameGenerator->reveal(), DummyProduct::class, 'get');

        $expected = <<<DQL
SELECT o
FROM ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\DummyProduct p
INNER JOIN p.offers offers
INNER JOIN p.relatedProducts relatedProducts
WHERE p IN(
  SELECT p
  FROM ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\DummyProduct p
  INNER JOIN p.offers offers
  INNER JOIN p.relatedProducts relatedProducts
  WHERE offers.value = :offerValue
  AND relatedProducts.name = :productName
)
DQL;

        $this->assertEquals($this->toDQLString($expected), $qb->getDQL());
    }
```